spring-cloud-azure-context/src/main/java/com/microsoft/azure/spring/cloud/context/core/util/Memoizer.java

```diff
@@ -34,4 +34,10 @@
     public static <T, U, R> BiFunction<T, U, R> memoize(Map<Tuple<T, U>, R> map, BiFunction<T, U, R> biFunction) {
         return (t, u) -> map.computeIfAbsent(Tuple.of(t, u), (k) -> biFunction.apply(k.getFirst(), k.getSecond()));
     }
+    
+    public static <T, U, R> BiFunction<T, U, R> memoizeCompute
```

Comment:
Try to understand your change. You try to create a new instance of EventProcessorHost even when old one existed. How about just changing the logic of unregister?

1. Add get method for this, rather than getOrCreate
2. Inside unregister, if existed, remove from cache, otherwise, do nothing.

Reply:
Dear Friend...
maybe only you have a deep knowledge of the code(functional behaviour)...I can suggest a non-invasive modification to allow the code works without crash. But, as already said, there are a strange behaviour that, frankly speaking, I didn't understand. During the unregister (into com.microsoft.azure.spring.integration.eventhub.impl.AbstractEventHubTemplate) even when a EventProcessorHost is null ...a new one is being created. So I cannot modify a great chunk of your code without having a whole picture of the functional behaviour. And moreover I cannot modify a substantial part of code without ask to you more info....it's common sense.

Reply:
Thanks anyway. May I close this? I'll handle the fix.

Reply:
> Try to understand your change. You try to create a new instance of EventProcessorHost even when old one existed. How about just changing the logic of unregister?
> 
> 1. Add get method for this, rather than getOrCreate
> 2. Inside unregister, if existed, remove from cache, otherwise, do nothing.

hi , i try to fix this bug. May i confirm that
1. Add remove method in EventHubClientFactory to remove EventProcessorHost from cache
2. As showed below, Inside unregister, if existed, remove from cache, otherwise, do nothing.
`protected void unregister(String name, String consumerGroup) {
        this.clientFactory.getOrCreateEventProcessorHost(name, consumerGroup).unregisterEventProcessor()
                          .whenComplete((s, t) -> {
                              if (t != null) {
                                  log.warn(String.format("Failed to unregister consumer '%s' with group '%s'", name,
                                          consumerGroup), t);
                              }else{
                                  this.clientFactory.removeEventProcessorHost(name, consumerGroup);
                              }
                          });
    }`

Reply:
Hi @warrenzhu25 , @chenghaoharvey .
You can check my take - following @warrenzhu25 feedback - on this bug @ #546